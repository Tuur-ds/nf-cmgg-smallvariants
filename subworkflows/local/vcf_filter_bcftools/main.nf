//
// Filter the VCFs
//

include { BCFTOOLS_FILTER as FILTER_1 } from '../../../modules/nf-core/bcftools/filter/main'
include { BCFTOOLS_FILTER as FILTER_2 } from '../../../modules/nf-core/bcftools/filter/main'
include { TABIX_TABIX                 } from '../../../modules/nf-core/tabix/tabix/main'

workflow VCF_FILTER_BCFTOOLS {
    take:
        ch_vcfs // channel: [ val(meta), path(vcf), path(tbi) ]
        val_tabix // boolean: whether to create a index or not

    main:

    ch_versions = Channel.empty()

    FILTER_1(
        ch_vcfs.map { meta, vcf, tbi=[] -> [ meta, vcf ]}
    )
    ch_versions = ch_versions.mix(FILTER_1.out.versions.first())

    FILTER_2(
        FILTER_1.out.vcf
    )
    ch_versions = ch_versions.mix(FILTER_2.out.versions.first())

    if(val_tabix) {
        TABIX_TABIX(
            FILTER_2.out.vcf
        )
        ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.first())

        FILTER_2.out.vcf
            .join(TABIX_TABIX.out.tbi, failOnDuplicate: true, failOnMismatch: true)
            .set { ch_filter_vcfs }
    } else {
        FILTER_2.out.vcf
            .set { ch_filter_vcfs }
    }


    emit:
    vcfs = ch_filter_vcfs  // channel: [ val(meta), path(vcf), path(tbi) ]

    versions = ch_versions // channel: [ versions.yml ]
}
