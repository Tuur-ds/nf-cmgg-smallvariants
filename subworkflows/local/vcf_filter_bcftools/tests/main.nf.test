nextflow_workflow {

    name "Test Workflow VCF_FILTER_BCFTOOLS"
    script "../main.nf"
    workflow "VCF_FILTER_BCFTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_filter_bcftools"

    test("vcf_filter_bcftools") {

        config "./nextflow.config"

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", caller:"haplotypecaller"],
                    file(params.vcf1, checkIfExists:true),
                    file(params.tbi1, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "variantsMD5:${path(it[1]).vcf.variantsMD5}"] }
                ).match() }
            )
        }

    }

}
