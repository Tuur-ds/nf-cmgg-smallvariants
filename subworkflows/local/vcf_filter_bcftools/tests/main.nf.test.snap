{
    "vcf_filter_bcftools": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "haplotypecaller"
                    },
                    "variantsMD5:2ce8bc96a9b3afbf060cdd89e74c4c82"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T17:11:49.839538282"
    }
}